src/app/dataset-view/datasetSubs.service.ts

```diff
@@ -0,0 +1,43 @@
+import {BehaviorSubject, Observable} from "rxjs";
+import {Injectable} from "@angular/core";
+import {SearchHistoryInterface, SearchOverviewInterface} from "../interface/search.interface";
+import {MetadataBlockFragment} from "../api/kamu.graphql.interface";
+@Injectable()
+export class AppDatasetSubsService {
+    private datasetOverviewChanges$: BehaviorSubject<SearchHistoryInterface[]> = new BehaviorSubject<SearchHistoryInterface[]>([]);
```

Comment:
Since we have `BehaviorSubject` here - my understanding is that it will cache the last value - i.e. it maintains a state.

If so, why do we need `SearchHistoryInterface[]` and similar variables inside of the `dataset.component.ts`? Can't we make individual table components (query result, metadata history, sample data etc.) subscribe directly to this state so that whenever it is updated components were updated as well?

Can you explain why `dataset.component.ts` needs to aggregate state for all of the individual tables?

Reply:
we need a parent component that will manage all the other tabs and tell where to write what. for presentation, we already have different components